secrets/awskms/kms.go

```diff
@@ -24,17 +34,96 @@ package awskms // import "gocloud.dev/secrets/awskms"
 import (
 	"context"
 	"errors"
+	"fmt"
+	"net/url"
+	"path"
+	"sync"
 
 	"github.com/aws/aws-sdk-go/aws/awserr"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/kms"
+	gcaws "gocloud.dev/aws"
 	"gocloud.dev/gcerrors"
 	"gocloud.dev/internal/gcerr"
 	"gocloud.dev/secrets"
 )
 
+func init() {
+	secrets.DefaultURLMux().RegisterKeeper(Scheme, new(lazySessionOpener))
+}
+
+// Dial gets an AWS KMS service client.
+func Dial(p client.ConfigProvider) (*kms.KMS, error) {
+	if p == nil {
+		return nil, errors.New("getting KMS service: no AWS session provided")
+	}
+	return kms.New(p), nil
+}
+
+// URLOpener opens secrets.Keeper URLs for AWS KMS, like "awskms://keyID".
+// The keyID can be in the form of an Amazon Resource Name (ARN), alias
+// name, or alias ARN. See
+// https://docs.aws.amazon.com/kms/latest/developerguide/viewing-keys.html#find-cmk-id-arn
+// for more details.
+// The URL Host + Path are used as the keyID, to support alias names like "awskms://alias/foo".
```

Comment:
> The URL Host + Path are used as the keyID

Here the keyID can be understood as either 1) the key resource ID this provider uses in the corresponding KMS, or 2) the resource ID we, the Go CDK, defined as an identifier which contains 1). It seems to be 2) and is worth clarifying somewhere. Maybe in the package doc.

Reply:
Sorry, what is the difference?

I intended this to be the AWS key ID, which can take several forms as discussed at the link pointed to above. They refer to it as "key ID", not "key resource ID". I'll update some comments to say "key ID" instead of "keyID" or "key resource ID" to be consistent with them.

I don't think there's a separate resource ID that we're defining.

Reply:
I think I had misunderstood that, I was thinking they can mean both our definition and provider's definition of key ID. But they are essentially the same thing. Thanks for making these consistent though, that could help.